src/Analyzers/CSharp/Analyzers/UsePatternMatching/CSharpUseNotPatternDiagnosticAnalyzer.cs

```diff
@@ -64,25 +65,32 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             if (!styleOption.Value)
                 return;
 
-            // Look for the form: !(x is Y y)
-            if (!(node is PrefixUnaryExpressionSyntax
+            // Look for the form: !(...)
+            if (node is not PrefixUnaryExpressionSyntax
                 {
-                    Operand: ParenthesizedExpressionSyntax
-                    {
-                        Expression: IsPatternExpressionSyntax
-                        {
-                            Pattern: DeclarationPatternSyntax,
-                        } isPattern,
-                    },
-                } notExpression))
+                    Operand: ParenthesizedExpressionSyntax parenthesizedExpression
+                } notExpression)
             {
```

Comment:
we also need to check that this is actually a `!` expression.  There are other prefix-unary expressions in the alngauge as well.

Reply:
it doesn't look like you use 'notExpression'.  can you remove if so?

Reply:
> I use it in `ReportDiagnostic` call at the end.

I think you can use `node` for that.